supabase/migrations/17_ops_catalogs.sql

```diff
@@ -0,0 +1,128 @@
+	ops_template json;
+	
+begin
+	select replace(bundled_catalog::text, 'TENANT', rtrim(tenant_prefix, '/'))::json into ops_template from ops_catalog_template where id =  '00:00:00:00:00:00:00:00';
+
+	insert into drafts (id, user_id, detail) values (new_draft_id, ops_user_id, 're-publishing ops catalog');
+	
+	-- Start by adding a deletion for all currently live ops catalog items.
+	insert into draft_specs (id, draft_id, catalog_name, spec_type, spec)
+		select
+			internal.id_generator() as id,
+			new_draft_id as draft_id,
+			catalog_name,
+			null as spec_type,
+			null as spec
+		from live_specs
+		where catalog_name like ('ops/' || tenant_prefix || '%');
+
+	insert into draft_specs (id, draft_id, catalog_name, spec_type, spec)
+		(
+		select
+			internal.id_generator() as id,
+			new_draft_id as draft_id,
+			"key" as catalog_name,
+			'capture'::catalog_spec_type as spec_type,
+			value as spec
+		from json_each(json_extract_path(ops_template, 'captures'))
+		union all select
+			internal.id_generator() as id,
+			new_draft_id as draft_id,
+			"key" as catalog_name,
+			'collection'::catalog_spec_type as spec_type,
+			value as spec
+		from json_each(json_extract_path(ops_template, 'collections'))
+		union all select
+			internal.id_generator() as id,
+			new_draft_id as draft_id,
+			"key" as catalog_name,
+			'materialization'::catalog_spec_type as spec_type,
+			value as spec
+		from json_each(json_extract_path(ops_template, 'materializations'))
+		union all select
+			internal.id_generator() as id,
+			new_draft_id as draft_id,
+			"key" as catalog_name,
+			'test'::catalog_spec_type as spec_type,
+			value as spec
+		from json_each(json_extract_path(ops_template, 'tests'))
+		)
+		on conflict (draft_id, catalog_name) do update
+			set spec_type = excluded.spec_type, spec = excluded.spec;
+
+	insert into publications (id, user_id, draft_id) values (publication_id, ops_user_id, new_draft_id);
+	return publication_id;
+end;
+$$ language plpgsql
+security definer;
+
+comment on function internal.create_ops_publication is '
+Creates a new publication of the ops catalog template for a specific tenant.
+This publication will include all specs from the bundled_catalog in ops_catalog_template.
+Any _other_ specs under the ops/<tenant>/ prefix will be deleted by this publication.
+';
+
+
+create function internal.update_ops_catalog_template(bundled_catalog_arg json)
+returns void as $$
+begin
+	insert into ops_catalog_template (id, bundled_catalog) 
+		values ('00:00:00:00:00:00:00:00', bundled_catalog_arg)
+		on conflict (id) do update set bundled_catalog = bundled_catalog_arg;
+	return;
+end;
+$$ language plpgsql
+security definer;
+
+comment on function internal.update_ops_catalog_template is '
+This function just makes migrations which update the ops catalog template look a little cleaner.
+It just upserts the bundled catalog into ops_catalog_template.
+';
+
+create function internal.republish_all_ops_catalogs(ops_user_id uuid)
```

Comment:
I also don't think this needs to be a function. As with `update_ops_catalog_template`, the _only_ place we expect to call it from is the migration script, so why not have it there? IMO not having this indirection would make the migration script itself more readable.

(I was thinking you had it here so you could test over it, but I'm not seeing that below)